### PR TITLE
DEVPROD-650 Remove deprecated version status

### PR DESCRIPTION
**Pull request by hadjri**
DEVPROD-650

### Description
Removing legacy version status and surrounding logic as followup from [EVG-19914](https://jira.mongodb.org/browse/EVG-19914)
### Testing
Existing tests

### CODE REVIEW

File: agent/command/git_merge_pr.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func (c *gitMergePR) Execute(ctx context.Context, comm client.Communicator, logg
 		return errors.Wrap(err, "setting up GitHub status logger")
 	}
 
-	if !evergreen.IsSuccessfulVersionStatus(patchDoc.MergeStatus) {
+	if !(patchDoc.MergeStatus == evergreen.VersionSucceeded) {
```
Comment (ZackarySantana) [inline agent/command/git_merge_pr.go:86]: I think we should just do != instead of wrapping with !()

File: model/project_ref.go [hunk 2]
```diff
@@ -3299,7 +3299,7 @@ func ValidateTriggerDefinition(definition patch.PatchTriggerDefinition, parentPr
 		return definition, errors.Wrapf(err, "finding child project '%s'", definition.ChildProject)
 	}
 
-	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.LegacyPatchSucceeded, evergreen.VersionSucceeded, evergreen.VersionFailed}, definition.Status) {
+	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.VersionSucceeded, evergreen.VersionFailed}, definition.Status) {
```
Comment (ZackarySantana) [inline model/project_ref.go:3302]: Not an action but would it be worth having some slice that encapsulates this? We don't add statuses (I think) so I think it's fine if this is the sole check but an idea

File: trigger/patch.go [hunk 3]
```diff
@@ -168,7 +168,7 @@ func finalizeChildPatch(sub *event.Subscription) error {
 }
 
 func (t *patchTriggers) patchSuccess(sub *event.Subscription) (*notification.Notification, error) {
-	if !evergreen.IsSuccessfulVersionStatus(t.data.Status) || t.event.EventType == event.PatchChildrenCompletion {
+	if !(t.data.Status == evergreen.VersionSucceeded) || t.event.EventType == event.PatchChildrenCompletion {
```
Comment (ZackarySantana) [inline trigger/patch.go:171]: I think we should do the same thing with != instead of !()

File: trigger/version.go [hunk 4]
```diff
@@ -240,7 +240,7 @@ func (t *versionTriggers) versionFailure(sub *event.Subscription) (*notification
 }
 
 func (t *versionTriggers) versionSuccess(sub *event.Subscription) (*notification.Notification, error) {
-	if !evergreen.IsSuccessfulVersionStatus(t.data.Status) || t.event.EventType == event.VersionChildrenCompletion {
+	if !(t.data.Status == evergreen.VersionSucceeded) || t.event.EventType == event.VersionChildrenCompletion {
```
Comment (ZackarySantana) [inline trigger/version.go:243]: ^ != thing

File: trigger/version.go [hunk 5]
```diff
@@ -301,7 +301,7 @@ func (t *versionTriggers) versionFamilyFailure(sub *event.Subscription) (*notifi
 }
 
 func (t *versionTriggers) versionFamilySuccess(sub *event.Subscription) (*notification.Notification, error) {
-	if !evergreen.IsSuccessfulVersionStatus(t.data.Status) || t.event.EventType != event.VersionChildrenCompletion {
+	if !(t.data.Status == evergreen.VersionSucceeded) || t.event.EventType != event.VersionChildrenCompletion {
```
Comment (ZackarySantana) [inline trigger/version.go:304]: ^ != thing

**Review by ZackarySantana — APPROVED** [review]
LGTM but I am but a secondary reviewer

File: model/project_ref.go (hunk 2 above)
Comment (ablack12) [reply model/project_ref.go:3302]: yeah this could be a function constant that helps us know what we're checking anyway, like `validTriggerStatuses`

**Review by ablack12 — COMMENTED** [review]
We're technically a month early on merging this but i think it's fineeeee

**Review by ablack12 — APPROVED** [review]

**Comment by hadjri** [comment]
evergreen refresh

**Pull request merged**